detection.py

```diff
@@ -226,11 +229,29 @@ def bridge_detection(self, image):
         if counter > 100:
             location = np.argwhere(binary_map == 0)
             f_x = np.mean(a=location, axis=0)[1]
-            mid = binary_map.shape[1] / 2
-            x_range = 0.02
-            if np.abs(f_x - mid) <= x_range * binary_map.shape[1]:
-                # print(f_x)
+            if np.abs(f_x - self.mid) <= self.x_range * binary_map.shape[1]:
+                return True
+        return False
+
+    def gate_detection(self, image):
+        '''
+        This algorithm gives whether we detect the gate\n
+        if gate is detected, return `True`, else return `false`
+        Written by Wen Bo, modified by Han Haoran
+        '''
+        # Get the  gradient map
+        gradient = cv2.Sobel(image, cv2.CV_64F, 1, 0, ksize =3)
```

Comment:
这里有点不一样, 参数列表里的等号两侧不建议有空格 :smirk:
```suggestion
        gradient = cv2.Sobel(image, cv2.CV_64F, 1, 0, ksize=3)
```